### ISSUE TITLE
getlog crashes lightningd

### ISSUE THREAD

**Issue by OttoAllmendinger** [ElementsProject/lightning#744]
This happened for `./cli/lightning-cli getlog io | grep '<peerid>'`

```
lightningd(23712): Connected json input
lightning_channeld: Writing out status 32768 len 27: Broken pipe
lightning_channeld: Writing out status 32768 len 27: Broken pipe
lightning_channeld: Writing out status 32768 len 27: Broken pipe
lightning_channeld: Writing out status 32768 len 27lightning_channeld: Writing out status 32768 len 27: Broken pipe
: Broken pipe
lightning_channeld: Writing out status 32768 len 27: Broken pipe
lightning_channeld: Writing out status 32768 len 27: Broken pipe
lightning_onchaind: Writing out status 32767 len 26: Broken pipe
lightning_channeld: lightning_onchaind: Writing out status 32767 len 26: Broken pipe
Writing out status 32768 len 27: Broken pipe
```

I saw it once when piping to `less` as well

commit 73c31757e18b0eaa6939e0be8e84302543f11760

on the first `lightningd` restart I hit this bug again: https://github.com/ElementsProject/lightning/issues/713

the second restart seems stable so far

**Comment by bjd** [comment]
A `getlog io` crash fix was added yesterday (see #726) so that will probably be gone if you can update to current head.

**Comment by OttoAllmendinger** [comment]
thanks, seems to have disappeared with latest master
